### ISSUE TITLE
changing pagehandler for the pages plugin breaks the friend collections tabs

### ISSUE THREAD

**Issue by propertunist** [beck24/pagehandler_hijack#3]
i noticed that the tabs on the friend collections pages (collections/owner/username) were broken on my site and i could see in the browser console that an ajax call to the address was resulting in a 404:

```
http://mysite.org/reference/friends/collections/pickercallback.php
```

the 'reference' handler is one i use on my site to replace the default 'pages' handler for the pages plugin. for some reason when i replace pages with reference, the change does not result in the friend collection pingercallback script being handled appropriately and the function fails.

any idea why?

**Comment by propertunist** [comment]
oh, so i recall/realise now. there are at least two uses of the handler 'pages' in elgg - the pages plugin is one and obviously for some reason, this context of the friend collections is another. it appears that perhaps the friend collections code doesn't register a normal page handler and is just hard coded to use 'pages' as the handler.. meaning that when i change it in pagehandler hijack, the address is not updated correctly for this context.

**Comment by propertunist** [comment]
the '_elgg_user_settings_page_handler()' function also lists hardcoded paths that start with 'pages', but i don't recall any problem with those.

**Comment by propertunist** [comment]
in fact, there are several other areas in the sourcecode that reference such paths, including the avatar edit path.. yet these do not have the same problems as the friend collections do.

**Comment by beck24** [comment]
I would consider this a core bug, there's no reason we should be using hard coded paths to php files anymore

**Comment by propertunist** [comment]
i agree that there should be no hardcoded paths (although there appear to be many)..
however, i am still not clear on why this issue is only effecting the friends picker and not the avatar edit page and others that use /pages/

**Comment by beck24** [comment]
Without trying this out myself I don't know offhand - it probably depends on how/where those urls are generated.  If they're used in js then they wouldn't be affected.  The root of the problem here is that the /pages/friends/collections/pickercallback.php is being treated as link/url but it doesn't actually route through a page handler.
I'd need to see the view where it's being used.

**Comment by propertunist** [comment]
the ones i have found are used within inline HTML script elements. e.g.  in /views/default/core/friends/collection.php
with a comment that says: 

```
//@todo JS 1.8: no 
```

i just realised that the reason for using the 'pages' path is that it is directly pointing to the pages folder in the root of the elgg installation (i missed that until now).
